src/go/rpk/pkg/cli/cmd/cluster/metadata.go

```diff
@@ -162,99 +162,165 @@ func printBrokers(controllerID int32, brokers kadm.BrokerDetails) {
 	}
 }
 
-func PrintTopics(topics kadm.TopicDetails, internal, detailed bool) {
-	if !detailed {
+// Used for structured output. Note: structured output is always detailed and doens't require the -d flag.
+type TopicPartitionDetail struct {
+	PartitionID        int32   `json:"partition_id" yaml:"partition_id"`                 // Partition is the partition number these details are for.
+	LeaderID           int32   `json:"leader_id" yaml:"leader_id"`                       // Leader is the broker leader  if there is one  otherwise -1.
+	LeaderEpoch        int32   `json:"leader_epoch" yaml:"leader_epoch"`                 // LeaderEpoch is the leader's current epoch.
+	ReplicaIDs         []int32 `json:"replica_ids" yaml:"replica_ids"`                   // Replicas is the list of replicas.
+	OfflineReplicasIDs []int32 `json:"offline_replicas_ids" yaml:"offline_replicas_ids"` // OfflineReplicas is the list of offline replicas.
+	Err                error   `json:"error" yaml:"error"`                               // Err is non-nil if the partition currently has a load error.
+}
+type Topic struct {
+	Name           string                 `json:"name" yaml:"name"`
+	PartitionCount int                    `json:"partition_count" yaml:"partition_count"`
+	Partitions     []TopicPartitionDetail `json:"partitions" yaml:"partitions"`
+	ReplicaCount   int                    `json:"replica_count" yaml:"replica_count"`
+	IsInternal     bool                   `json:"is_internal" yaml:"is_internal"`
+}
+type Topics struct {
+	Topics []Topic `json:"topics" yaml:"topics"`
+}
+
+func (collection *Topics) sortTopics() {
+	sort.Slice(collection.Topics, func(i, j int) bool {
+		l, r := collection.Topics[i], collection.Topics[j]
```

Comment:
l, r := &..., &...

Reply:
I believe this is my inexperience with go. This worked as it was written before, and works the same (visibly at least) with the suggested change, but I assume with out using `&` the runtime was copying the topics to l and r the `l, r :=` statement vs using pointers so this is saves cpu/memory overhead? The suggested change will go in regardless, but I want to make sure I'm not making bad assumptions about why the suggestion was made.